stock_available_to_promise_release/models/stock_move.py

```diff
@@ -145,3 +174,18 @@ def _run_stock_rule(self):
             pulled_moves = pulled_moves.mapped("move_orig_ids")
 
         return True
+
+    def _release_split(self, remaining_qty):
+        """Split move and create a new picking for it.
+
+        Instead of splitting the move and leave remaining qty into the same picking
+        we move it to a new one so that we can release it later as soon as
+        the qty is available.
+        """
+        # Rely on `printed` flag to make _assign_picking create a new picking.
+        # See `stock.move._assign_picking` and
+        # `stock.move._search_picking_for_assignation`.
+        self.picking_id.printed = True
+        new_move = self.browse(self._split(remaining_qty))
+        new_move._assign_picking()
```

Comment:
is the call to `_assign_picking()` necessary? I would have thought `_split()` would call `_action_confirm()` which in turn calls `_assign_picking` (but maybe the conditions to call it are not fulfilled so you need to call it manually).

Reply:
ehh no, this is what we thought but I found that that flow happens only if you have moves to be assigned, which is not the case here. 
See
https://github.com/odoo/odoo/blob/f662c1582eb55a7f7fc1e8c0a3eea0e519f1aa64/addons/stock/models/stock_move.py#L1013
and 
https://github.com/odoo/odoo/blob/f662c1582eb55a7f7fc1e8c0a3eea0e519f1aa64/addons/stock/models/stock_move.py#L1034
Picking and picking type are already there because the move is copied.

Reply:
:+1: alternatively, we could have emptied the `picking_id` with an override of `_prepare_move_split_vals` (https://github.com/odoo/odoo/blob/f662c1582eb55a7f7fc1e8c0a3eea0e519f1aa64/addons/stock/models/stock_move.py#L1442) when we have the `release_available_to_promise` context key, but calling `_assign_picking` should work as well.

Reply:
I added a comment